blog/2023-08-01-tolgee-with-next-13.md

```diff
@@ -0,0 +1,347 @@
+---
+slug: tolgee-with-nextjs-app-router
+title: 'Tolgee with Next.js app router'
+authors: [sgranat]
+description: 'Setting up Tolgee with Next.js new app router and server components.'
+tags: [next.js, server components, react]
+---
+
+With the introduction of Next.js 13 and the app router featuring React server components, there has been a strong push to enable Tolgee to work within this new paradigm.
+
+Until now, Tolgee was designed to function with regular React components. However, we now face the challenge of making in-context translation work with server components. Let's delve into the details.
+
+<!--truncate-->
+
+> #### How Tolgee In-Context Works
+>
+> Tolgee's unique method of enabling users to directly edit translations within the app is based on inserting invisible characters next to the actual translations. This creates a kind of `watermark` for each translation, making it detectable in the DOM and allowing precise localization.
+> This approach offers a non-intrusive way to empower users to translate in the context of their app.
+
+### The General Plan
+
+For server components, the approach involves including a complete key name within the invisible `watermark` characters. Additionally, the development translation files need to be loaded on the server to facilitate rendering. On the client side, the SDK will recognize the server-rendered `watermarks` and enable the in-context translation functionality.
+
+# Setting Up the Configuration
```

Comment:
So it's actually a tutorial? I think you should split this to 2 articles. Tutorial with click-bait title, second where you can explain how it works and what was the challenges. That one might be more into how the server components work etc.